app/controllers/api/v1/notifications/requests_controller.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+class Api::V1::Notifications::RequestsController < Api::BaseController
+  before_action -> { doorkeeper_authorize! :read, :'read:notifications' }, only: :index
+  before_action -> { doorkeeper_authorize! :write, :'write:notifications' }, except: :index
+
+  before_action :require_user!
+  before_action :set_request, except: :index
+
+  after_action :insert_pagination_headers, only: :index
+
+  def index
+    with_read_replica do
+      @requests = load_requests
+      @relationships = StatusRelationshipsPresenter.new(@requests.map(&:last_status), current_user&.account_id)
+    end
+
+    render json: @requests, each_serializer: REST::NotificationRequestSerializer, relationships: @relationships
+  end
+
+  def accept
+    AcceptNotificationRequestService.new.call(@request)
+    render_empty
+  end
+
+  def dismiss
+    @request.update!(dismissed: true)
+    render_empty
+  end
+
+  private
+
+  def load_requests
+    requests = NotificationRequest.where(account: current_account, dismissed: false).includes(:last_status, from_account: [:account_stat, :user]).to_a_paginated_by_id(
```

Comment:
Instead of local var, pull this whole thing thing out to private method?

Reply:
I don't see anything to be gained from that. It's also consistent with the way a similar method is written in the notifications controller.